src/coins.h

```diff
@@ -155,75 +106,10 @@ class CCoins
         return !(a == b);
     }
 
-    void CalcMaskSize(unsigned int &nBytes, unsigned int &nNonzeroBytes) const;
-
     bool IsCoinBase() const {
         return fCoinBase;
     }
 
-    template<typename Stream>
-    void Serialize(Stream &s) const {
```

Comment:
My rationale behind this deletion is that the serialization described here is not really a property of the coins type; it is instead a specialized property of how this data is serialized for the database. This is unlike, for example, the CTransaction serialization, where its serialized form is canonical and protocol-relevant. Basically, since this logic belongs to txdb exclusively, I think that the code should continue to reside there. Otherwise you get a bit of an odd tangling of concerns; this way if we choose to replace the data store and require somewhat different serialization, the new serialization code will be on equal footing with and can coexist with the current implementation.